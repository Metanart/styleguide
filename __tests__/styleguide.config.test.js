import StyleguideConfig from '../styleguide.config';
import path from 'path';

describe('Config checks', () => {
    it('Config Structure', () => {
        expect(StyleguideConfig).toMatchSnapshot();
    });

    it('Config getExceptionForLoaders', () => {
        expect(StyleguideConfig.getExceptionForLoaders()).toMatchSnapshot();
    });

    it('Config getSections', () => {
        try {
            expect(StyleguideConfig.getSections()).toMatchSnapshot();
        } catch (err) {
            console.log(err);
        }
    });

    it('Config getComponentRoots', () => {
        expect(StyleguideConfig.getComponentRoots({ path })).toMatchSnapshot();
    });

    it('Config getWebpackConfig', () => {
        expect(StyleguideConfig.getWebpackConfig({ path })).toMatchSnapshot();
    });
});
