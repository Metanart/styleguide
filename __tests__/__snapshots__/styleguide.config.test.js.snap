// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config checks Config Structure 1`] = `
Object {
  "getComponentRoots": [Function],
  "getExceptionForLoaders": [Function],
  "getSectionComponents": [Function],
  "getSections": [Function],
  "getWebpackConfig": [Function],
  "hasResizableSandbox": false,
  "sandboxMaxWidth": undefined,
  "sandboxMinWidth": undefined,
}
`;

exports[`Config checks Config getComponentRoots 1`] = `
Array [
  "/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/src",
]
`;

exports[`Config checks Config getExceptionForLoaders 1`] = `
Object {
  "jsLoader": Array [
    /spec\\\\\\.jsx\\$/,
  ],
}
`;

exports[`Config checks Config getSectionComponents 1`] = `
Array [
  Object {
    "components": Array [
      "/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/src/components/component/component",
    ],
  },
  Object {
    "components": Array [
      "/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/src/components/error-boundary/error-boundary",
    ],
  },
]
`;

exports[`Config checks Config getWebpackConfig 1`] = `
Object {
  "resolve": Object {
    "modules": Array [
      "/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/src",
      "/Users/igrekov/Downloads/_exp/task08-styleguide/styleguide/node_modules",
    ],
  },
}
`;
